jinhwan/H2309/user.cpp

```diff
@@ -0,0 +1,105 @@
+#include <algorithm>
+#include <vector>
+#include <unordered_map>
+#include <map>
+
+#define MAX_N 100001
+#define MAX_M 1001
+
+using namespace std;
+
+int sectionTime[MAX_N] = {0,};
+int segmentTree[MAX_N*4];
+int sectionType[MAX_N][MAX_M];
+
+unordered_map<int, vector<int>> typeSections;
+
+int N, M;
+
+void build(int node, int start, int end)
+{
+    if(start == end)
+    {
+        segmentTree[node] = sectionTime[start];
+    }else
+    {
+        int mid = (start + end) / 2;
+        build(2*node, start, mid);
+        build(2*node + 1, mid + 1, end);
+        segmentTree[node] = segmentTree[2*node] + segmentTree[2*node + 1];
+    }
+}
+
+void updateTree(int node, int start, int end, int mID, int diff)
+{
+    if(mID < start || mID > end) return;
+    if(start == end)
+    {
+        if(mID == start) segmentTree[node] += diff;
+    }
+    else{
+        segmentTree[node] += diff;
+        int mid = (start + end) / 2;
+        updateTree(2*node, start, mid, mID, diff);
+        updateTree(2*node + 1, mid + 1, end, mID, diff);
+    }
```

Comment:
```suggestion
    if(mID < start || mID > end) return;
    if(start == end)
    {
        segmentTree[node] += diff;
        return;
    }
    
    segmentTree[node] += diff;
    int mid = (start + end) / 2;
    updateTree(2*node, start, mid, mID, diff);
    updateTree(2*node + 1, mid + 1, end, mID, diff);
```

이거 사실 취향차이고, 사소한 부분이긴 한데 난 early return을 선호하는 편!
Ref. [🙅🏻‍♀️ 니들은 else 같은거 쓰지마라](https://velog.io/@gomjellie/else-%EC%93%B0%EC%A7%80%EB%A7%88)